vm/src/stdlib/signal.rs

```diff
@@ -98,15 +98,21 @@ fn _signal_alarm(time: u32) -> u32 {
 }
 
 #[cfg_attr(feature = "flame-it", flame)]
+#[inline(always)]
 pub fn check_signals(vm: &VirtualMachine) -> PyResult<()> {
-    let signal_handlers = match vm.signal_handlers {
-        Some(ref h) => h.borrow(),
+    let signal_handlers = match &vm.signal_handlers {
+        Some(h) => h,
         None => return Ok(()),
     };
 
     if !ANY_TRIGGERED.swap(false, Ordering::Relaxed) {
         return Ok(());
     }
+
+    trigger_signals(&signal_handlers.borrow(), vm)
+}
+#[inline(never)]
```

Comment:
How about `cold` attribute? this is a question instead of suggestion because i haven't seen any actual usage of `cold` yet. https://doc.rust-lang.org/reference/attributes/codegen.html

Reply:
That sounds good, I was thinking that it really isn't *that* exceptional of a case, but given that it is (possibly) called for every instruction execution it would make sense to signal that it should very rarely ever happen.